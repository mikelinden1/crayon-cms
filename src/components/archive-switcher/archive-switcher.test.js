import ArchiveSwitcher from './archive-switcher';
import sinon from 'sinon';

const props = {
    archiveMode: 'all',
    config: {
        itemName: 'test', 
        itemNamePlural: 'tests',
        archive: {
            field: 'start_date',
            archiveText: 'Archive',
            nonArchiveText: 'Future'
        }
    },
    actions: {
        setArchive: () => {}
    }
}

it('ArchiveSwitcher matches snapshot', () => {
    const wrapper = shallow(<ArchiveSwitcher {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('ArchiveSwitcher doesn\'t display if we don\'t have an archive config', () => {
    const propsNoArchive = {
        ...props,
        config: {
            archive: null
        }
    };

    const wrapper = shallow(<ArchiveSwitcher {...propsNoArchive} />);
    expect(wrapper.type()).toEqual(null)
});

it('ArchiveSwitcher button text matches config', () => {
    const wrapper = mount(<ArchiveSwitcher {...props} />);

    const allBtn = wrapper.find('Button').first().text();
    expect(allBtn).toEqual(`All ${props.config.itemNamePlural}`);

    const nonArchiveBtn = wrapper.find('Button').at(1).text();
    expect(nonArchiveBtn).toEqual(props.config.archive.nonArchiveText);

    const archiveBtn = wrapper.find('Button').at(2).text();
    expect(archiveBtn).toEqual(props.config.archive.archiveText);
});

it('ArchiveSwitcher calls setArchive and sends new archive state', () => {
    const spy = sinon.spy();
    const propsWithSpy = {
        ...props,
        actions: {
            setArchive: spy
        }
    };

    const wrapper = mount(<ArchiveSwitcher {...propsWithSpy} />);
    wrapper.find('Button').at(1).simulate('click');

    expect(spy.calledOnce).toBeTruthy();
    expect(spy.getCall(0).args[0]).toEqual('active');
});