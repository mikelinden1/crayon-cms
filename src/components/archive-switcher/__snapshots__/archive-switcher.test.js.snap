// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArchiveSwitcher matches snapshot 1`] = `
<div
  className="btn-group"
  style={
    Object {
      "border": "1px solid #1a1a1a",
    }
  }
>
  <Button
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    All 
    tests
  </Button>
  <Button
    color="secondary"
    onClick={[Function]}
    tag="button"
  >
    Future
  </Button>
  <Button
    color="secondary"
    onClick={[Function]}
    tag="button"
  >
    Archive
  </Button>
</div>
`;
